sonar-batch/src/main/java/org/sonar/batch/source/DefaultHighlightable.java

```diff
@@ -32,15 +33,17 @@
 
   private final DefaultInputFile inputFile;
   private final SensorStorage sensorStorage;
+  private final AnalysisMode analysisMode;
 
-  public DefaultHighlightable(DefaultInputFile inputFile, SensorStorage sensorStorage) {
+  public DefaultHighlightable(DefaultInputFile inputFile, SensorStorage sensorStorage, AnalysisMode analysisMode) {
     this.inputFile = inputFile;
     this.sensorStorage = sensorStorage;
+    this.analysisMode = analysisMode;
   }
 
   @Override
   public HighlightingBuilder newHighlighting() {
-    DefaultHighlighting defaultHighlighting = new DefaultHighlighting(sensorStorage);
```

Comment:
Wouldn't it be better if the builder builds a no-op object (NoOpHighlighting), leaving DefaultHighlighting unchanged?